### ISSUE TITLE
Bulk "create" via Java API (1.4.2) doesn't error on duplicate docs but does return negative version

### ISSUE THREAD

**Issue by Alex-Ikanow** [elastic/elasticsearch#11685]
I am creating via BulkRequestBuilder, doing

    add(client.prepareIndex(...).setOpType(OpType.CREATE)

I have a test where I bulk insert and a subset of the objects inserted already exist. I then attach a listener to the bulk request, and expect to get .hasFailures() etc. (This is running vs an in memory server created in JUnit, if that makes any difference)

Instead what happens is hasFailures() returns false, and the inserts silently fail (ie they appear in the requests() array with isFailed()==false, but aren't updated). 

One difference I do see in the listener is that the new objects have version==1 whereas the failed indexes have ==-3, eg (my duplicate objects start at "id51"):

    WORKED2: 29...id50...CREATE...{"_id":"id50","test_string":"test_string250"}...1
    WORKED2: 30...id51...CREATE...{"_id":"id51","test_string":"test_string251"}...-3

(for

    System.out.println("WORKED2: " + bir.getItemId() + "..." + bir.getId() + "..." + ir.opType().toString() + "..." + ir.source().toUtf8() + "..." + ir.version());
)

I confirm the object is not updated in the DB, eg:

    ---------------- {"_id":"id50","test_string":"test_string250"} // (newly inserted object)
    ---------------- {"_id":"id51","test_string":"test_string51"} // (original object)

Any ideas what's going on here?

(EDIT: note if i change from OpType.CREATE to OpType.INDEX then as expected I see versions 1 for new objects, 2 for updated objects, and the objects are updated in the DB)

Is this because I'm using a listener and there's a second wave of verification that occurs after the onResponse()? I'm "happy" to use negative version to indicate "duplicate insert failure" if that's going to be consistent behavior.

(Apologies if this behavior is explained somewhere, I couldn't find it!)

**Comment by clintongormley** [comment]
Hi @Alex-Ikanow 

Could you include the actual code that you use?

**Comment by Alex-Ikanow** [comment]
Ugh file this one under "can't reproduce/operator error", apologies - I pulled out the code into a standalone method and (of course!) it worked as expected (version is still -3; but isFailed() is now correctly returning true, and getFailureMessage() is returning the correct message instead of null as before). 

I then tried my original test inside my more complicated code (which has been modified since the original report) and this time it appeared to report the duplicates correctly. 

I think it's safe to assume I was doing something stupid... apologies again!
